### PR TITLE
feat: Add OMG support

### PR DESCRIPTION
**Pull request by wsdt**
- The Gateway shows the OMG token as option once we have called the addSupportedToken function on the smart contracts. 

ToDo's: 
- Add OMG as supported route on smart contracts
- Provide OMG liquidity to multisigs/disburser wallets 

aborted as per
https://enyainc.slack.com/archives/C06TJKR0H0R/p1713382756677069

### CODE REVIEW

File: test/lightbridge.integration.spec.ts [hunk 1]
```diff
@@ -2136,6 +2137,134 @@ describe('lightbridge', () => {
       )
     })
   })
+
+  describe('OMG support', () => {
+      before(async () => {
+        const Factory_OMGLike = new ethers.ContractFactory(
+          L1ERC20Json.abi,
+          L1ERC20Json.bytecode,
+          signer
```
Comment (souradeep-das) [inline test/lightbridge.integration.spec.ts:2146]: you didn't use OMGLikeToken contract?

File: test/lightbridge.integration.spec.ts [hunk 2]
```diff
@@ -2136,6 +2137,134 @@ describe('lightbridge', () => {
...
+        utils.parseEther('10')
+      )
+
+      const amountDisbursements = await LightBridge.connect(
+        signer
+      ).totalDisbursements(chainId)
+
+      expect(amountDisbursements).to.be.eq(1)
+    })
+
```
Comment (souradeep-das) [inline test/lightbridge.integration.spec.ts:2266]: can we also add a test for the scenario - where the recipient is not able to  receive OMG and in that case the disbursement call reverts as usual (just to double check this return value nuance of the OMG contract)

File: test/lightbridge.integration.spec.ts (hunk 1 above)
Comment (wsdt) [reply test/lightbridge.integration.spec.ts]: great catch!!

File: test/lightbridge.integration.spec.ts [hunk 3]
```diff
@@ -2207,63 +2190,150 @@ describe('lightbridge', () => {
...
+      ]
+      expect(token.toLowerCase()).to.be.eq(OMGLikeToken.address.toLowerCase())
+
+      disbursement = orderBy(disbursement, ['depositId'], ['asc'])
+
+      const preOMGBalance = await OMGLikeToken.balanceOf(address1)
+      const preSignerOMGBalance = await OMGLikeToken.balanceOf(signerAddr)
+
+      console.log("Pausing lightbridge")
+      let pauseTx = await LightBridge.connect(wallet1).pause()
```
Comment (souradeep-das) [inline test/lightbridge.integration.spec.ts:2313]: woops, I meant intentionally failing the internal "token transfer" triggered. I think making this line intentionally fail  https://github.com/bobanetwork/light-bridge/blob/main/contracts/LightBridge.sol#L316 could help verify this. Maybe we just call disburse without omg token approvals and that helps revert this line for us?

**Review by souradeep-das — APPROVED** [review]

File: test/lightbridge.integration.spec.ts (hunk 3 above)
Comment (wsdt) [reply test/lightbridge.integration.spec.ts:2313]: done! :-)

**Review by sk-enya — APPROVED** [review]

**Review by souradeep-das — APPROVED** [review]

**Pull request closed without merging**